newdocs/typeclass-cheatsheet.md

```diff
@@ -0,0 +1,223 @@
+# Consensus Cheatsheet (Type Families, Classes, & Notes)
+
+See [Diagrammatic Conventions](#diagrammatic-conventions)
+    
+## Birds Eye Overview of Consensus Types
+### Type Families (standalone and associated types)
+    
+From inspecting the directions of the arrows, it should be clear that `b :: B`
+fully determines---directly or indirectly---all the other types.
+
+``` haskell
+ P[rotocol]                       B[lock]                       L[edger]                     -- (the P,B,L "kinds")
+===                              ===                           ===
+                                                                                                  ┏━━━━━━━━━━━┓
+ p  ──(ConsensusConfig :: P→*)──────────────────────────────────────────────────────────────▶ cc  ┃{- static-}┃
+ p  ◀──(BlockProtocol :: B→P)──── b ──(BlockConfig :: B→*)──────────────────────────────────▶ bc  ┃{- config-}┃
+                                  b ──(LedgerState :: B→L)──▶ l ──(LedgerCfg :: L→*)────────▶ lc  ┃{- data  -}┃
+                                                                                                  ┗━━━━━━━━━━━┛
+                                  
+                                  b ──(LedgerState :: B→L)──▶ l ──(AuxLedgerEvent :: L→*)──▶ lev   -- events emitted by ledger
+                                                              l ──(LedgerErr :: L→*)───────▶ lerr  -- errors when updating ledger
+
+
+                                  b ──(CodecConfig   :: B→*)──────▶ codecc -- for serialisation and deserialisation
+                                  b ──(StorageConfig :: B→*)──────▶ sc     -- other config [?]
```

Comment:
I think you could replace this stub comment with `for (re)initializing the Storage Layer`

Reply:
done